pkg/server/registry/apigroups/acorn/events/strategy.go

```diff
@@ -0,0 +1,80 @@
+package events
+
+import (
+	"context"
+	"sort"
+	"strconv"
+
+	apiv1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	"github.com/acorn-io/mink/pkg/strategy"
+	"github.com/acorn-io/mink/pkg/types"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+type eventStrategy struct {
+	strategy.CompleteStrategy
+}
+
+func (s *eventStrategy) List(ctx context.Context, namespace string, opts storage.ListOptions) (types.ObjectList, error) {
+	// Unmarshal custom field selectors and strip them from the list options before
+	// passing to lower-level strategies (that don't support them).
+	// TODO(njhale): I'm sure there's a better way to (un)marshal these
+	var q query
+	stripped, err := opts.Predicate.Field.Transform(func(f, v string) (string, string, error) {
+		var err error
+		switch f {
+		case "details":
+			q.details, err = strconv.ParseBool(v)
+		default:
+			return f, v, nil
+		}
+
+		return "", "", err
+	})
+	if err != nil {
+		return nil, err
+	}
+	opts.Predicate.Field = stripped
+	logrus.Warnf("query: [%v]", q)
+
+	// TODO(njhale): Filtering like this adds an extra O(n*lgn) time and O(n) space to every List call.
```

Comment:
For now this is fine. I think we can optimize if this becomes a major scaling factor for us that'll be a contained segment of work when we do so.

Reply:
I concur, I dropped the comment.